src/test_l3_dsFPD.c

```diff
@@ -0,0 +1,627 @@
+*
+* This test case provides a scope to  get the Color of Given indicator
+*
+* **Pre-Conditions:** @n
+* Front panel Module should be intialized through Test 1 before calling this test.
+* Front panel indicator state should be set to ON through Test 2 before calling this test
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 10 and provide indicator index.
+*
+*/
+void test_10_dsFPD_hal_GetFPColor(void)
+{
+    gTestID = 10;
+    UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t status = dsERR_NONE;
+    dsFPDColor_t color;
+
+    selectIndicator();
+    UT_LOG_INFO("Calling dsGetFPColor(IN:Indicator:[0x%d], OUT:Color)",eIndicator);
+    status = dsGetFPColor((dsFPDIndicator_t)eIndicator,&color);
+    UT_LOG_INFO("Result dsGetFPColor(IN:Indicator:[0x%d], OUT:Color:[%d]) dsError_t:[%s]" \
+                                                ,eIndicator,color,UT_Control_GetMapString(dsFrontPanelErrorCodeTable, status));
+    assert(status == dsERR_NONE);
+    /* Check that the Indicator is valid */
+
+    UT_LOG_INFO("Out %s\n", __FUNCTION__);
+}
+/**
+* @brief This test provides a scope to Termnate Device Settings Front panel Module.
+*
+* This test case provides a scope to terminate front panel module.
+*
+* **Pre-Conditions:** @n
+* Front panel Module should be intialized through Test 1 before calling this test.
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 11.
+*
+*/
+
+void test_11_dsFPD_hal_Term(void)
+{
+   gTestID = 11;
+   dsError_t status = dsERR_NONE;
+
+   UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+   // Step 1: Call dsFPInit()
+   UT_LOG_INFO("Calling dsFPTerm()");
+   status = dsFPTerm();
+   UT_LOG_INFO("Result dsFPTerm()dsError_t:[%s]",UT_Control_GetMapString(dsFrontPanelErrorCodeTable, status));
+   assert(status == dsERR_NONE);
+
+   UT_LOG_INFO("Out %s\n", __FUNCTION__);
+}
+
+static UT_test_suite_t * pSuite = NULL;
+
+/**
+ * @brief Register the main tests for this module
+ *
+ * @return int - 0 on success, otherwise failure
+ */
+int test_register_dsFPD_hal_l3_tests(void)
+{
+    // Create the test suite
+    pSuite = UT_add_suite("[L3 Front Panel Functions] ", NULL, NULL);
+    if (pSuite == NULL)
+    {
+        return -1;
+    }
+    // List of test function names and strings
+
+    UT_add_test( pSuite, "1_Init_dsFPD", test_1_dsFPD_hal_Init);
+    UT_add_test( pSuite, "2_SetState_dsFPD", test_2_dsFPD_hal_SetFPState);
+    UT_add_test( pSuite, "3_GetState_dsFPD", test_3_dsFPD_hal_GetFPState);
+    UT_add_test( pSuite, "4_SetBlink_dsFPD", test_4_dsFPD_hal_SetFPBlink);
+    UT_add_test( pSuite, "5_SetBrightness_dsFPD", test_5_dsFPD_hal_SetFPBrightness);
+    UT_add_test( pSuite, "5_GetBrightness_dsFPD", test_6_dsFPD_hal_GetFPBrightness);
+    UT_add_test( pSuite, "7_SetFPState_dsFPD", test_7_dsFPD_hal_SetLEDState);
```

Comment:
I would prefer setLEDState and getLEDstate rather than setFPState and getFPState to be in sync with the API name.

Reply:
updated